src/main/java/net/test/interceptor/StackTraceInterceptor.java

```diff
@@ -31,22 +32,34 @@ public String toString() {
 
 public class StackTraceInterceptor {
 
+  private static String LOG_THRESHOLD_MILLISECONDS = "log_threshold_ms";
+
   private static List<String> KNOWN_ARGS = 
-    Arrays.asList(CommonActionArgs.IS_DATE_LOGGED);
+    Arrays.asList(CommonActionArgs.IS_DATE_LOGGED, LOG_THRESHOLD_MILLISECONDS);
 
   private CommonActionArgs commonActionArgs;
 
+  private final long logThresholdMs;
+
   public StackTraceInterceptor(String actionArgs, DefaultArguments defaults) {
-    Map<String, String> parsed = ArgUtils.parseOptionalArgs(KNOWN_ARGS, actionArgs);
+    ArgumentsCollection parsed = ArgUtils.parseOptionalArgs(KNOWN_ARGS, actionArgs);
     this.commonActionArgs = new CommonActionArgs(parsed, defaults);
+    this.logThresholdMs = parsed.parseLong(LOG_THRESHOLD_MILLISECONDS, 0);
   }
 
   @RuntimeType
   public Object intercept(@Origin Method method, @SuperCall Callable<?> callable) throws Exception  {
-    Exception e = new MyException(commonActionArgs.addPrefix("TraceAgent (stack trace)"));
-    StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
-    e.setStackTrace(Arrays.copyOfRange(stElements, 2, stElements.length));
-    e.printStackTrace(System.out);
-    return callable.call();
+    long start = System.currentTimeMillis();
```

Comment:
Let's avoid calling `System.currentTimeMillis()` when the threshold is 0 (for both start and end) for all the interceptors.

Reply:
I made the change for `TraceArgsInterceptor` and `StackTraceInterceptor`. For the timing interceptors `TimingInterceptorMs` and `TimingInterceptorNano` the implementation already required both start and end calls to `System.currentTimeMillis()` hence haven't done any changes.